### PR TITLE
Fix Tab Control Text and add more TabControl tests

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- This was never set in #2258 

Contributes to https://github.com/dotnet/winforms/issues/2462

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2465)

### CODE REVIEW

**Comment by hughbe** [comment]
This is *not* a problem in 3.1 /cc @russkie

**Comment by hughbe** [comment]
Got some tests to validate - incoming

**Comment by RussKie** [comment]
> This is _not_ a problem in 3.1

Correct, all refactors are on the master branch.

**Comment by RussKie** [comment]
Tests appears to dead lock on the CI.

Running locally I get multiple locking popups with the following error:
```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.ArgumentException: 'child' is not a child control of this parent.
   at System.Windows.Forms.Control.ControlCollection.GetChildIndex(Control child, Boolean throwException) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\Control.ControlCollection.cs:line 484
   at System.Windows.Forms.Control.ControlCollection.GetChildIndex(Control child) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\Control.ControlCollection.cs:line 468
   at System.Windows.Forms.Control.UpdateChildZOrder(Control ctl) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\Control.cs:line 11670
   at System.Windows.Forms.Control.WmCreate(Message& m) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\Control.cs:line 12110
   at System.Windows.Forms.Control.WndProc(Message& m) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\Control.cs:line 13137
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\ScrollableControl.cs:line 1452
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\Control.ControlNativeWindow.cs:line 65
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\Control.ControlNativeWindow.cs:line 117
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WindowMessage msg, IntPtr wparam, IntPtr lparam) in C:\Development\winforms\src\System.Windows.Forms\src\System\Windows\Forms\NativeWindow.cs:line 375


************** Loaded Assemblies **************
....
```

I can also observe one or two tests spawning forms and those dead lock as well.
I didn't have much time to investigate.

**Comment by weltkante** [comment]
> I can also observe one or two tests spawning forms and those dead lock as well.

Might be the same ones I observed in #2192 - because they don't respond to window messages they are deadlocking any test which involves sending global window messages to all forms.

It requires looking at the test and/or deadlocked stack traces though, if it doesn't broadcast a window message then the forms are just visible but aren't the cause of this particular deadlock.

The origin of those forms is probably some test which didn't dispose properly of forms it created. Its kinda hard to identify the offending tests though.

**Comment by hughbe** [comment]
test hang were caused by #2234

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Properties.cs [hunk 1]
```diff
@@ -6023,6 +6023,54 @@ public void Control_Location_SetWithHandleWithTransparentBackColor_DoesNotCallIn
             Assert.Equal(expectedInvalidatedCallCount * 2, invalidatedCallCount);
         }
 
+        [WinFormsFact]
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Properties.cs]: How do these relate to testing the fix?

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Properties.cs]: TabPage overrides ShouldSerializeLocation so I was testing the default behaviour of Location serialization

Comment (JeremyKuhne) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Properties.cs]: Can we add a test for the regression?

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Properties.cs]: Please keep only tests related to the fix, and submit all other tests separately.

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Properties.cs]: Ok. Tests for the fix added - all of them fail before this change

**Pull request merged**